spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDStreamParser.java

```diff
@@ -354,20 +385,27 @@ public DocumentParseResult parse(String[] document) {
 			DocumentParseResult result = new DocumentParseResult(document.length);
 			CrudRepository<BaseDefinition, String> transientRepository = new TransientDefinitionRepository();
 			StreamConfigParser parser = new StreamConfigParser(transientRepository);
+			int line = 1;
 			for (String nameAndDefinitionPair : document) {
 				try {
-
 					StreamNode stream = parser.parse(nameAndDefinitionPair);
 					String streamName = stream.getStreamName();
+					if (streamName == null) {
+						// Give the stream a placeholder name as none was supplied. Someone
+						// processing the parse result should know this can happen.
+						streamName = "UNKNOWN_" + line;
+					}
+					List<Exception> errorAccumulator = new ArrayList<Exception>();
 					List<ModuleDescriptor> moduleDescriptors = delegate.buildModuleDescriptors(streamName,
-							nameAndDefinitionPair, ParsingContext.stream, stream);
+							nameAndDefinitionPair, ParsingContext.partial_stream, stream, errorAccumulator);
 					BaseDefinition streamDefinition = new StreamDefinition(streamName, nameAndDefinitionPair);
 					transientRepository.save(streamDefinition);
-					result.success(moduleDescriptors);
+					result.success(moduleDescriptors, errorAccumulator);
```

Comment:
ok, thank you for clarifying. Now I understand we only handle `NoSuchModuleException` and the syntax error. The module option validation is not in place as we need positional info for that.